src/common/FileSystem.cpp

```diff
@@ -1059,8 +1056,20 @@ static void InternalLoadPak(const PakInfo& pak, Util::optional<uint32_t> expecte
 			return;
 	}
 
-	// Add the pak to the list of loaded paks
-	fsLogs.Notice("Loading pak '%s'...", pak.path.c_str());
+	if (pak.type == pakType_t::PAK_ZIP) {
+		if (!isLegacy) {
+			fsLogs.Notice("Loading pak '%s'...", pak.path.c_str());
+		} else {
+			fsLogs.Notice("Loading legacy pak '%s'...", pak.path.c_str());
+		}
+	} else if (pak.type == pakType_t::PAK_DIR) {
+		if (!isLegacy) {
+			fsLogs.Notice("Loading pakdir '%s'...", pak.path.c_str());
+		} else {
+			fsLogs.Notice("Loading legacy pakdir '%s'...", pak.path.c_str());
+		}
+	}
```

Comment:
@Kangz do you think I have to add an `else UNREACHABLE()` there?
In fact everywhere else the code is just doing `if PAK_DIR then else `(and does not test for `PAK_ZIP`).
I can also modify the other occurrence with explicit test for `PAK_ZIP` when not `PAK_DIR`, plus an `else UNREACHABLE`. What do you think about it?